packages/gatsby-plugin-sharp/src/utils.js

```diff
@@ -88,8 +88,27 @@ export function rgbToHex(red, green, blue) {
     .slice(1)}`
 }
 
+const checkIgnoredParameters = (layout, parameters, filepath) => {
+  const ignoredParams = Object.entries(parameters).filter(([_, value]) =>
+    Boolean(value)
+  )
+  if (ignoredParams.length) {
+    console.warn(
+      `The following provided parameter(s): ${ignoredParams
+        .map(param => param.join(`: `))
+        .join(
+          `, `
+        )} for the image at ${filepath} are ignored in ${layout} image layouts.`
+    )
+  }
+  return
+}
+
+const DEFAULT_PIXEL_DENSITIES = [0.25, 0.5, 1, 2, 3]
```

Comment:
We should not enable density 3 by default, for constrained images it hardly make sense unless you have a gallery where you have thumbnails and a zoom-in picture. I would add a "highRes" prop to enable this behavior instead.

A good article about this https://blog.twitter.com/engineering/en_us/topics/infrastructure/2019/capping-image-fidelity-on-ultra-high-resolution-devices.html

```suggestion
const DEFAULT_PIXEL_DENSITIES = [0.25, 0.5, 1, 2]
```

Reply:
Rather than a highRes prop, we let the user pass in an array of densities. Agreed that the default should be a max of 2 though

Reply:
highRes is food for thought cause it might be easier for people to understand that than srcBreakPoints. Again, not for now ;)